code/modules/cargo/centcom_podlauncher.dm

```diff
@@ -91,103 +157,115 @@
+		if("loadDataFromPreset")
+			var/list/savedData = params["payload"]
+			loadData(savedData)
 			. = TRUE
-		if("bay5")
-			bay =  locate(/area/centcom/supplypod/loading/ert) in GLOB.sortedAreas
+		if("switchBay")
+			bayNumber = params["bayNumber"]
 			refreshBay()
 			. = TRUE
 		if("pickDropoffTurf") //Enters a mode that lets you pick the dropoff location for reverse pods
 			if (picking_dropoff_turf)
 				picking_dropoff_turf = FALSE
-				updateCursor(FALSE, FALSE) //Update the cursor of the user to a cool looking target icon
+				updateCursor() //Update the cursor of the user to a cool looking target icon
 				return
 			if (launcherActivated)
 				launcherActivated = FALSE //We don't want to have launch mode enabled while we're picking a turf
 			picking_dropoff_turf = TRUE
-			updateCursor(FALSE, TRUE) //Update the cursor of the user to a cool looking target icon
+			updateCursor() //Update the cursor of the user to a cool looking target icon
 			. = TRUE
 		if("clearDropoffTurf")
 			picking_dropoff_turf = FALSE
-			dropoff_turf = null
-			updateCursor(FALSE, FALSE)
+			customDropoff = FALSE
+			var/turf/T = pick(bay)
+			temp_pod.reverse_dropoff_coords = list(T.x, T.y, T.z)
+			indicator.forceMove(T)
+			updateCursor()
 			. = TRUE
-		if("teleportCentcom") //Teleports the user to the centcom supply loading facility.
+		if("teleportDropoff") //Teleports the user to the dropoff point.
 			var/mob/M = holder.mob //We teleport whatever mob the client is attached to at the point of clicking
-			oldTurf = get_turf(M) //Used for the "teleportBack" action
-			var/area/A = locate(bay) in GLOB.sortedAreas
-			var/list/turfs = list()
-			for(var/turf/T in A)
-				turfs.Add(T) //Fill a list with turfs in the area
-			if (!length(turfs)) //If the list is empty, error and cancel
-				to_chat(M, "Nowhere to jump to!")
-				return //Only teleport if the list isn't empty
-			var/turf/T = pick(turfs)
+			var/turf/current_location = get_turf(M)
+			var/list/c = temp_pod.reverse_dropoff_coords
+			var/turf/T = locate(c[1],c[2],c[3])
+			if (current_location != T)
+				oldTurf = current_location
 			M.forceMove(T) //Perform the actual teleport
 			log_admin("[key_name(usr)] jumped to [AREACOORD(T)]")
 			message_admins("[key_name_admin(usr)] jumped to [AREACOORD(T)]")
 			. = TRUE
-		if("teleportBack") //After teleporting to centcom, this button allows the user to teleport to the last spot they were at.
+		if("teleportCentcom") //Teleports the user to the centcom supply loading facility.
+			var/mob/M = holder.mob //We teleport whatever mob the client is attached to at the point of clicking
+			var/turf/current_location = get_turf(M)
+			var/area/A = bay
+			if (current_location.loc != A)
+				oldTurf = current_location
+			var/turf/T = pick(A)
+			M.forceMove(T) //Perform the actual teleport
+			if (holder.holder)
+				log_admin("[key_name(usr)] jumped to [AREACOORD(T)]")
+				message_admins("[key_name_admin(usr)] jumped to [AREACOORD(T)]")
```

Comment:
```suggestion
			var/mob/holder_mob = holder.mob //We teleport whatever mob the client is attached to at the point of clicking
			var/turf/current_location = get_turf(holder_mob)
			var/area/bay_area = bay
			if (current_location.loc != bay_area)
				oldTurf = current_location
			var/turf/teleport_turf = pick(bay_area)
			holder_mob.forceMove(teleport_turf) //Perform the actual teleport
			if (holder.holder)
				log_admin("[key_name(usr)] jumped to [AREACOORD(teleport_turf)]")
				message_admins("[key_name_admin(usr)] jumped to [AREACOORD(teleport_turf)]")
```
Are you sure this is right? `var/turf/teleport_turf = pick(bay_area)` ?
`pick()` from an area-typed variable?
In part this is why more clear variables are better to have.

Reply:
I've seen this be done elsewhere in the codebase. I figured it was some BYOND magic that automatically `pick()`ed a turf from the area. Can change it to something more clear